client/blocks/props-viewer/index.jsx

```diff
@@ -0,0 +1,192 @@
+	static displayName = 'PropsViewer';
+
+	constructor( props ) {
+		super( props );
+	}
+
+	/**
+	 * Finds a non-example component in the library of components
+	 * @param {string} slug The slug to search for
+	 * @return {Array} An array of component matches
+	 */
+	findRealComponent = ( slug ) => {
+		// remove the last character. As of right now, all plural display names are with just an 's'
+		const singular = slug.slice( 0, -1 );
+		return components.filter( ( component ) => {
+			return ( slug === component.slug || singular === component.slug ) && component.includePath.indexOf( 'example' ) < 0;
+		} );
+	};
+
+	/**
+	 * Set the state of this component to the first matching slug
+	 * @param {string} slug The slug to search for
+	 */
+	setComponent = ( slug ) => {
+		let component = this.findRealComponent( slug );
+		if ( component.length > 0 ) {
+			component = component[ 0 ];
+		} else {
+			component = null;
+		}
+
+		this.setState( {
+			component
+		} );
+	};
+
+	componentWillMount() {
+		this.setComponent( this.props.component );
+	}
+
+	componentWillReceiveProps( nextProps ) {
+		this.setComponent( nextProps.component );
+	}
+
+	/**
+	 * Sort prop names by (required, name)
+	 * @param {string} leftName The prop name of the left side
+	 * @param {string} rightName The prop name of the right side
+	 * @return {number} Which side wins
+	 */
+	sortProps = ( leftName, rightName ) => {
+		const component = this.state.component;
+
+		const left = component.props[ leftName ];
+		const right = component.props[ rightName ];
+		if ( left.required === right.required ) {
+			return ( leftName.toLowerCase() < rightName.toLowerCase() ? -1 : 1 );
+		}
+
+		if ( left.required ) {
+			return -1;
+		}
+
+		return 1;
+	};
+
+	/**
+	 * Renders a row in the table
+	 * @param {object} component The component
+	 * @param {string} propName The name of the prop to render
+	 * @return {ReactElement} The rendered row
+	 */
+	renderRow = ( component, propName ) => {
+		const prop = component.props[ propName ];
+		let type = 'unknown';
+		if ( prop.type ) {
+			switch ( prop.type.name ) {
+				default:
+					type = prop.type.name;
+					break;
+				case 'arrayOf':
+				case 'oneOf':
+				case 'oneOfType':
+				case 'objectOf':
+				case 'instanceOf':
+					type = `${ prop.type.name }( ${ prop.type.value.name || 'unknown' } )`;
+					break;
+			}
+		}
+
+		return (
+			<tr key={ propName }>
+				<td>{ prop.required ? <Gridicon icon="checkmark" /> : <Gridicon icon="cross-small" /> }</td>
+				<td>{ propName }</td>
+				<td>{ type }</td>
+				<td>{ prop.defaultValue ? prop.defaultValue.value : 'undefined' }</td>
+				<td>{ prop.description }</td>
+			</tr>
+		);
+	};
+
+	/**
+	 * Renders a table if it can
+	 * @param {object} component The component to render for
+	 * @return {ReactComponent|null} The table or nothing
+	 */
+	renderTable = ( component ) => {
+		if ( ! component ) {
+			return null; //todo: explain why this table is missing
+		}
+
+		return (
```

Comment:
It does feel a bit long, but it's mostly just markup. Lines 155 - 160 could probably become more readable.